Sources/KituraWebSocket/WSConnectionUpgradeFactory.swift

```diff
@@ -38,6 +41,26 @@ public class WSConnectionUpgradeFactory: ProtocolHandlerFactory {
         return connection
     }
 
+    // Return all the extension handlers enabled for this connection
+    public func extensionHandlers(header: String) -> [ChannelHandler] {
+        var handlers: [ChannelHandler] = []
+        for _extension in header.split(separator: ",") {
+            guard let name = _extension.split(separator: ";").first, let impl = extensions[String(name)] else { continue }
+            handlers += impl.handlers(header: String(_extension))
+        }
+        return handlers
+    }
+
+    // Let all the configured extensions negotiate for themselves, build a response header and send it back
+    public func negotiate(header: String) -> String {
```

Comment:
If this needs to be public we should have /// for jazzy docs. If this doesn't need to be public we can remove `public`.

Reply:
Done. Thanks!